mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -37,6 +37,42 @@ import org.apache.spark.sql.types._
  */
 private[feature] trait RFormulaBase extends HasFeaturesCol with HasLabelCol {
 
+  /**
+   * Param for how to order categories of a string FEATURE column used by `StringIndexer`.
+   * The last category after ordering is dropped when encoding strings.
+   * Supported options: 'frequencyDesc', 'frequencyAsc', 'alphabetDesc', 'alphabetAsc'.
+   * The default value is 'frequencyDesc'. When the ordering is set to 'alphabetDesc', `RFormula`
+   * drops the same category as R when encoding strings.
+   *
+   * The options are explained using an example `'b', 'a', 'b', 'a', 'c', 'b'`:
+   * {{{
+   * +-----------------+---------------------------------------+----------------------------------+
```

Comment:
@felixcheung @HyukjinKwon The scaladoc complied, but the javadoc failed...  Not sure if there is additional config for java? 

![image](https://cloud.githubusercontent.com/assets/11082368/26341144/048b8d6e-3f47-11e7-8600-c111643a0295.png)